cds/modules/deposit/api.py

```diff
@@ -49,9 +50,46 @@
 )
 
 
+class CDSFileObject(FileObject):
+    """Wrapper for files."""
+
+    def dumps(self):
+        """Create a dump of the metadata associated to the record."""
+        self.data.update({
+            'key': self.obj.key,
+            'version_id': str(self.obj.version_id),
+            'checksum': self.obj.file.checksum,
+            'size': self.obj.file.size,
+            'completed': True,
+            'progress': 100,
+            'links': {
+                'self': (
+                    current_app.config['DEPOSIT_FILES_API'] +
+                    u'/{bucket}/{key}?versionId={version_id}'.format(
+                        bucket=self.obj.bucket_id,
+                        key=self.obj.key,
+                        version_id=self.obj.version_id,
+                    )),
+            }
+        })
+        return self.data
+
+
 class CDSDeposit(Deposit):
     """Define API for changing deposit state."""
 
+    def __init__(self, *args, **kwargs):
+        self.file_cls = CDSFileObject
```

Comment:
I think this should have been a class attribute instead, see https://github.com/inveniosoftware/invenio-records-files/blob/master/invenio_records_files/api.py#L259